src/zenml/pipelines/pipeline_definition.py

```diff
@@ -656,7 +662,8 @@ def _create_deployment(
                 schedule_name = schedule.name
             else:
                 schedule_name = format_name_template(
-                    deployment.run_name_template
+                    deployment.run_name_template,
+                    **(deployment.pipeline_configuration.substitutions or {}),
```

Comment:
I played with this back in day, but it might be causing the lost of the values, if nothing is passed. E.g. substitutions are `{'a':'b'}` and empty update will make it `{}`. Thus kept in sync with other configs with None defaults

Reply:
If I look at `PipelineConfigurationUpdate` though, both `extra` and `settings` are non-optional though? And I'm sure we don't lose any values in empty updates

Reply:
did so